mitodl/open-discussions
# sort by department coursenum when there is a department filter

#### Pre-Flight checklist

#### What are the relevant tickets?
https://github.com/mitodl/open-discussions/issues/3453
RFC https://docs.google.com/document/d/1zEXSDFAJlOvtZmilUjKkwD-4P8d3fk1mIHQLcgMhFew/edit

#### What's this PR do?
This pr displays and sorts by the course number from the selected department when there is a department filter.

#### How should this be manually tested?
Make sure the migration works backwards and forwards

Run
```
sync_ocw_courses(course_prefixes=["PROD/6/6.901/Fall_2005/6-901-inventions-and-patents-fall-2005/"], blocklist=[], force_overwrite=True, upload_to_s3=False)
```
To verify that backpopulating ocw data isn't broken


Run:
```
    
from course_catalog.models import *

for course in Course.objects.filter(platform="ocw").iterator():
    course_run = course.runs.filter(raw_json__isnull=False, published=True).last()

    if course_run:
        course.extra_course_numbers = [
            extra_course_number_json["linked_course_number_col"]
            for extra_course_number_json in (
                course_run.raw_json.get("extra_course_number") or []
            )
            if extra_course_number_json.get("linked_course_number_col")
        ]
    

        course.save()
```
To backpopulate the data without reimporting all ocw courses. I will run this script on RC and Prod as well when this PR is approved.


Recreate the index

Run the main branch in ocw-hugo-themes with this branch for the index. Verify that everything works as expected

Run [ab/department-course-number-sort](https://github.com/mitodl/ocw-hugo-themes/pull/134) in ocw-hugo-themes with this branch for the index

Filter to department=Physics
Verify that every course number in the results starts with an `8` and when you select "sort by course number" they are sorted correctly.

Remove the filter. Search for "Inventions and Patents". The course should have number "6.901". Now filter to department ="Materials Science and Engineering". The course number should now be 3.172.

Clear the filters. Search for "3.172".  "Inventions and Patents" should be the first result
